### PR TITLE
feat(core, product): rename DaffSortable to DaffOrderable

### PR DESCRIPTION
**Pull request by damienwebdev**
BREAKING CHANGE: the interface name changed, the above swap is sufficient to breakfix.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Currently, we have an existing interface called "DaffSortable". This name will collide with the refactor on-going in DaffCategory. 

Fixes: N/A

## What is the new behavior?
In order to prevent the collision, I'm proposing "DaffOrderable" instead.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: libs/core/src/orderable/orderable.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * An interface for models where the order of elements is important
+ * and definable.
+ *
+ * Disambiguation: This is does not relate to ecommerce "Orders", e.g.a purchase.
```
Comment (griest024) [inline libs/core/src/orderable/orderable.ts:5]: missing space after `e.g.`

**Review by griest024 — CHANGES_REQUESTED** [review]

**Comment by lderrickable** [comment]
What does `DaffSortable` collide with in the category refactor? 

`Orderable` seems a little ambiguous in an e-commerce library, because it could mean that you can buy it or that you can sort it. So I think "sortable" is preferable, but I don't know what the particular collision is.

**Comment by lderrickable** [comment]
The only alternative I can think of is `DaffPositioned` with the field being `position` rather than `order`. I think I actually like `DaffOrderable` better with the disambiguity comment though

File: libs/core/src/orderable/orderable.ts (hunk 1 above)
Comment (damienwebdev) [reply libs/core/src/orderable/orderable.ts]: 13ac9d76

**Review by griest024 — APPROVED** [review]

**Pull request merged**